lib/rouge/lexers/supercollider.rb

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class SuperCollider < RegexLexer
+      tag 'supercollider'
+      filenames '*.sc', '*.scd'
+
+      title "SuperCollider"
+      desc 'A cross-platform interpreted programming language for sound synthesis, algorithmic composition, and realtime performance'
+
+      keywords = Set.new %w(
```

Comment:
We're trying to avoid the use of local variables for keywords as these result in object allocation just from the loading of the lexer. To that end, could you replace these assignments so that rather than this:

```rb
keywords = Set.new %w(example keywords)
```

you do it like this:

```rb
def self.keywords
  @keywords ||= Set.new %w(example keywords)
end
```

This will mean that later in the code, you'll need to refer to these as `self.class.keywords` rather than as `keywords`.